compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -5124,6 +5124,7 @@ object Types extends TypeUtils {
       }
 
       def isUpToDate: Boolean =
+        (reductionContext ne null) &&
```

Comment:
!= null is fine, it's the same code generated.

Reply:
Except values of type `MutableMap[Type, Type]` and `Null` cannot be compared with `!=`. 

Is there an advantage to doing 
```scala
private var reductionContext: util.MutableMap[Type, Type] = uninitialized
def setReductionContext(): Unit =
  ...
  reductionContext = util.HashMap()
 ```
rather than 
```scala
private var reductionContext: util.MutableMap[Type, Type] = util.HashMap()
def setReductionContext(): Unit =
  ...
  reductionContext.clear()
```
?

Reply:
I overlooked that. The proper thing to do is to declare reductionContext as 

```scala
    private var reductionContext: util.MutableMap[Type, Type] | Null = null
```
Generally, `= unintialized` should only be used if we never need to test for null afterwards since the logic ensures that the var is always assigned before use.